src/openmc_nek_driver.cpp

```diff
@@ -252,12 +252,43 @@ void OpenmcNekDriver::init_volumes()
   }
 }
 
-void OpenmcNekDriver::init_elem_densities()
+void OpenmcNekDriver::init_densities()
 {
   if (this->has_global_coupling_data()) {
-    elem_densities_.resize({gsl::narrow<std::size_t>(n_global_elem_)});
+    densities_.resize({gsl::narrow<std::size_t>(n_global_elem_)});
+    densities_prev_.resize({gsl::narrow<std::size_t>(n_global_elem_)});
+
+    if (density_ic_ == Initial::neutronics) {
+      // Loop over the OpenMC cells, then loop over the global Nek elements
+      // corresponding to that cell and assign the OpenMC cell density to
+      // the correct index in the densities_ array. This mapping assumes that
+      // each Nek element is fully contained within an OpenMC cell, i.e. Nek
+      // elements are not split between multiple OpenMC cells.
+      for (int i = 0; i < openmc_driver_->cells_.size(); ++i) {
+        if (cell_fluid_mask_[i] == 1) {
+          auto& c = openmc_driver_->cells_[i];
+          const auto& global_elems = mat_to_elems_.at(c.material_index_);
+
+          for (int elem: global_elems) {
+            double rho = c.get_density();
+            densities_[elem] = rho;
+            densities_prev_[elem] = rho;
+          }
+        }
```

Comment:
When initializing `densities_` from OpenMC, it seems that the densities of the the non-fluid elements  are undefined values:
* First, `densities_` is resized (which I believe doesn't initialize the new xtensor elements?).
* Then, the non-fluid elements (where `cell_fluid_mask[i] != 1`) are not initialized in this conditional.  

It shouldn't affect the current functionality, since we are using the fluid mask consistently.  But I'd prefer that the density for non-fluid elements be some well-defined value, in case we decide to rely on it later.  

When initializing `densities_` from Nek, we've implemented `NekDriver::density()` to set the densities of the non-fluid elements to 0.  Maybe we can follow the same convention here, too.

Reply:
Good point, I agree that the densities for solid elements should at least be consistent. I'll set them to zero